dvc/path/factory.py

```diff
@@ -39,7 +44,21 @@ def to_data_items(self, files):
             try:
                 data_item = DataItem(file, self._git, self._config)
                 result.append(data_item)
-            except NotInDataDirError:
+            except DataDirError:
                 externally_created_files.append(file)
 
         return result, externally_created_files
+
+    def all_existing_data_items(self):
+        items = []
+        states = []
+
+        for root, dirs, files in os.walk(self._config.state_dir):
+            for fname in files:
```

Comment:
Should we check `STATE_FILE_SUFFIX ` in this loop? It can help to avoid file name discrepancies in the next loop.

Reply:
How about a warning (or even an exception) if the suffix does not match?

Reply:
Agreed on both points =), will fix. Thanks a lot for review!